### DISCUSSION TITLE
9주차 과제

### DISCUSSION THREAD

**Discussion by youngsangcho** [youngsangcho/generativedesign2_2021#33] [Show and tell]
#과제
Dynamic Typography 코드 완성!!!
+ A-Z, BACKSPACE, ENTER 기능 구현
+ 쉼표, 마침표, 따옴표, 큰따옴표 디자인 추가 및 구현
+ 인터랙션 / 애니메이션 / 랜덤화 구현

+ 폰트 이름 정하기
+ 폰트 작업 설명 - 80자 이상, 120자 이내
+ 페이지 배경색 고르기 (#FF0000)
+ 자동으로 쓰여질 영어 문구. 30자 이상 

11월 22일 밤 12시까지.
더 일찍 제출하고 많이 질문할 수록 많은 도움을 받을 수 있음!

**Comment by SeyounJung01** [comment]
정세윤 https://blog.naver.com/seyounjj2001/222575944258

**Comment by Joeun410** [comment]
김조은
https://blog.naver.com/PostView.naver?blogId=joanne3456&Redirect=View&logNo=222575954893&categoryNo=21&isAfterWrite=true&isMrblogPost=false&isHappyBeanLeverage=true&contentLength=20718

**Comment by KimHyunJuhg** [comment]
김현정
https://blog.naver.com/khj317888/222575956473

**Comment by imyeeunok** [comment]
https://blog.naver.com/010312kye/222575957351

김예은

**Comment by eunjinhong** [comment]
홍은진
https://blog.naver.com/tnekf1012/222575957661

**Comment by jwejwepark** [comment]
https://blog.naver.com/glqkfl0726/222576187640

**Comment by KimHyunJuhg** [comment]
김현정
https://blog.naver.com/khj317888/222577683525

**Reply by youngsangcho** [reply]
'-' b

**Reply by KimHyunJuhg** [reply]
교수님 제가 기호 부분 코드 조금 수정해서 올렸는데 지금은 수정이 불가할까요!

**Comment by eunjinhong** [comment]
홍은진
https://blog.naver.com/tnekf1012/222578011819

**Reply by youngsangcho** [reply]
1. Type constructor의 아래 부분을 고쳐보세요.
this.keyCode =code;
    this.char = String.fromCharCode(code);
    if (code == 49) this.char = '!'; 
    if (code == 188) this.char = ',';
    if (code == 189) this.char = '-';
    if (code == 190) this.char = '.';
    if (code == 191) this.char = '?';
    if (code == 222) this.char = '\'';

2. json 에서 따옴표가 "''" 로 되어있던데 "'" 로 바뀌어야 합니다.

3. 색은 저녁까지 좀 더 고민해봐도 됩니다.

'-' b

**Comment by Joeun410** [comment]
김조은 
https://blog.naver.com/PostView.naver?blogId=joanne3456&Redirect=View&logNo=222578020249&categoryNo=23&isAfterWrite=true&isMrblogPost=false&isHappyBeanLeverage=true&contentLength=12995

**Reply by youngsangcho** [reply]
'-' b

**Comment by SeyounJung01** [comment]
https://blog.naver.com/seyounjj2001/222578051495

**Reply by youngsangcho** [reply]
1. 코드 맨 아래   } else 지워면 오류 해결
2. 77~80번 줄 때문에 코드 느려지는 듯 합니다. 1초에 수십개씩 새로운 버튼을 만들어 추가하고 있어서 부하가 걸리는 것 같네요. 그 부분 모두 지워도 될 것 같습니다.

**Reply by SeyounJung01** [reply]
https://editor.p5js.org/jsypaul2001/sketches/aTNN4_IES
수정했습니다! 
혹시 남은 시간이 가능하다면 제가 상단바를 코드 맨 아래쪽으로 옮기는 코드를 짜봐도 될까요?

**Reply by youngsangcho** [reply]
네, 근데 위에 있는 것도 좋아보이는데? 아래 있으면 잘 안 보이지 않을까요?

**Reply by SeyounJung01** [reply]
글자 크기를 키우면 상단바를 침범하는 문제 때문이였는데 생각보다 쉽게 해결했습니다..(혹시 버튼의 스타일도 다른글씨처럼 바꾸는 방법이 있을까요? ) ++ 처음 재생버튼을 누르면 글자가 중첩되는지 매우 굵어졌다가 키보드를 누르면 다시 얇아지는 현상이 있는데, 이는 어떻게 고칠 방법이 있을까요?
https://editor.p5js.org/jsypaul2001/sketches/QebAoYzuf

**Reply by youngsangcho** [reply]
이 부분에 아래처럼 noStroke() 추가해주면 됩니다. text 그릴 때 스트로크가 그려져서 그렇습니다.
이 상태로 올리도록 하겠습니다. 수고했어요.

noStroke();

fill("#6624e1");
  text("Click : rotate", 15, fontY);
  text("Esc : reset", 135, fontY);
  text("⇅ key : space size", 235, fontY);
  text("↔ key : wave size", 395, fontY);
  text("[ ] key : font size", 555, fontY);
  text("Mousewheel : cursor size", 715, fontY);
  text("space : ", 965, fontY);
  text(spaceW, 1020, fontY);
  text("wave : ", 1065, fontY);
  text(waveS / 5, 1115, fontY);
  text("font size", 1160, fontY);
  text(jung, 1230, fontY);

**Comment by imyeeunok** [comment]
https://blog.naver.com/010312kye/222578068242

**Reply by youngsangcho** [reply]
'-' b

**Comment by jwejwepark** [comment]
https://editor.p5js.org/juhye/sketches/477Ot1kk7

(마우스 x값으로 글자들이 움직이게 조절했는데, 글씨들이 한곳에 다 모였을때 이미지(배경)이 어두워지는걸 시도했으나 잘 안되었다.)

**Reply by jwejwepark** [reply]
https://blog.naver.com/glqkfl0726/222578121945
(블로그 링크 적는걸까먹었습니다.)

**Reply by youngsangcho** [reply]
'-' b

**Reply by jwejwepark** [reply]
페이지 배경색 255.0.0 레드 부탁드립니다.감사합니다

**Reply by youngsangcho** [reply]
넵
